### PR TITLE
[fuzzer] Populate arrays and enums

### PR DESCRIPTION
**Pull request by tonkosi**

### CODE REVIEW

File: tools/fuzzer/symbol_table.cc [hunk 1]
```diff
@@ -180,6 +205,17 @@ int calculate_freedom_index(lldb::SBValue value,
     }
   }
 
+  if (type.IsArrayType()) {
+    lldb::addr_t address =
+        static_cast<lldb::addr_t>(value.AddressOf().GetValueAsUnsigned());
+    if (is_valid_address(address, memory_regions)) {
+      // The first array element.
+      lldb::SBValue element_value = value.CreateValueFromAddress(
+          "value", address, type.GetArrayElementType());
```
Comment (werat) [inline tools/fuzzer/symbol_table.cc:214]: Maybe it's better to use `SBValue::GetChildAtIndex` for this? For arrays it returns the element at specified index.

File: tools/fuzzer/symbol_table.cc [hunk 2]
```diff
@@ -344,24 +380,37 @@ void load_frame_variables(SymbolTable& symtab, lldb::SBFrame& frame,
...
+      lldb::eTypeClassStruct | lldb::eTypeClassClass |
+      lldb::eTypeClassEnumeration);
   uint32_t types_size = types.GetSize();
 
   for (uint32_t i = 0; i < types_size; ++i) {
     lldb::SBType type = types.GetTypeAtIndex(i);
-    if (!is_tagged_type(type)) {
-      continue;
+
+    // Structs and classes
```
Comment (werat) [inline tools/fuzzer/symbol_table.cc:396]:
```suggestion
    // Structs and classes.
```

File: tools/fuzzer/symbol_table.cc [hunk 3]
```diff
@@ -344,24 +380,37 @@ void load_frame_variables(SymbolTable& symtab, lldb::SBFrame& frame,
...
+        }
+      }
     }
-    const auto tagged_type = TaggedType(type.GetName());
-    const auto& info = classes.get_class_info(type);
-    for (const auto& field : info.fields()) {
-      if (info.is_unique_field_name(field.name)) {
-        symtab.add_field(tagged_type, field.name, field.type);
+
+    // Enumerations
```
Comment (werat) [inline tools/fuzzer/symbol_table.cc:407]:
```suggestion
    // Enumerations.
```

For consistency :)

File: tools/fuzzer/symbol_table_test.cc [hunk 4]
```diff
@@ -296,3 +322,55 @@ TEST_F(PopulateSymbolTableTest, TaggedTypesAndFields) {
...
+
+  {
+    const EnumType type("ns::EnumClass", kHasScopedEnums);
+    auto enum_it = symtab_.enums().find(type);
+    ASSERT_NE(enum_it, symtab_.enums().end());
+    const auto& literals = enum_it->second;
+
+    EXPECT_THAT(literals, UnorderedElementsAre(
+                              EnumConstant(type, "ns::EnumClass::ZERO"),
+                              EnumConstant(type, "ns::EnumClass::ONE"),
```
Comment (werat) [inline tools/fuzzer/symbol_table_test.cc:372]: Should `UnorderedElementsAre` be on the next line, same as in the code above? Weird that `clang-format` doesn't complain

Comment (tonkosi) [reply tools/fuzzer/symbol_table_test.cc:372]: If it is the same as in the code above, then the `ns::EnumClass::THREE` line will be 82 characters long.

Comment (werat) [reply tools/fuzzer/symbol_table_test.cc:372]: Oh I see, didn't notice that

**Review by werat — APPROVED** [review]

**Pull request merged**